src/widgets/font_description.rs

```diff
@@ -0,0 +1,115 @@
+            Some(FontDescription { pointer: tmp })
+        }
+    }
+
+    pub fn equal(desc1: &FontDescription, desc2: &FontDescription) -> bool {
+        unsafe { to_bool(ffi::pango_font_description_equal(desc1.pointer, desc2.pointer)) }
+    }
+
+    pub fn set_family(desc: &mut FontDescription, family: &str) {
+        unsafe { ffi::pango_font_description_set_family(desc.pointer, family.to_glib_none().0) };
+    }
+
+    pub fn set_family_static(desc: &mut FontDescription, family: &'static str) {
+        unsafe {
+            ffi::pango_font_description_set_family_static(desc.pointer, family.to_glib_none().0)
+        };
+    }
+
+    pub fn get_family(desc: &FontDescription) -> Option<String> {
+        unsafe {
+            let tmp = ffi::pango_font_description_get_family(desc.pointer);
+            from_glib_none(tmp)
+        }
+    }
+
+    pub fn set_weight(desc: &mut FontDescription, weight: enums::Weight) {
+        unsafe { ffi::pango_font_description_set_weight(desc.pointer, weight) };
+    }
+
+    pub fn get_weight(desc: &FontDescription) -> enums::Weight {
+        unsafe { ffi::pango_font_description_get_weight(desc.pointer) }
+    }
+
+    pub fn set_size(desc: &mut FontDescription, size: i32) {
+        unsafe { ffi::pango_font_description_set_size(desc.pointer, size) };
+    }
+
+    pub fn get_size(desc: &FontDescription) -> i32 {
+        unsafe { ffi::pango_font_description_get_size(desc.pointer) }
+    }
+
+    pub fn set_absolute_size(desc: &mut FontDescription, size: f64) {
+        unsafe { ffi::pango_font_description_set_absolute_size(desc.pointer, size) };
+    }
+
+    pub fn get_size_is_absolute(desc: &FontDescription) -> bool {
+        unsafe { to_bool(ffi::pango_font_description_get_size_is_absolute(desc.pointer)) }
+    }
+
+    pub fn set_gravity(desc: &mut FontDescription, gravity: enums::Gravity) {
+        unsafe { ffi::pango_font_description_set_gravity(desc.pointer, gravity) };
+    }
+
+    pub fn get_gravity(desc: &FontDescription) -> enums::Gravity {
+        unsafe { ffi::pango_font_description_get_gravity(desc.pointer) }
+    }
+
+    pub fn get_set_fields(desc: &FontDescription) -> enums::FontMask {
+        unsafe { ffi::pango_font_description_get_set_fields(desc.pointer) }
+    }
+
+    pub fn unset_fields(desc: &mut FontDescription, to_unset: enums::FontMask) {
+        unsafe { ffi::pango_font_description_unset_fields(desc.pointer, to_unset) };
+    }
+
+    pub fn from_string(name: &str) -> Option<FontDescription> {
+        let c_name = CString::new(name).unwrap();
+        let tmp = unsafe { ffi::pango_font_description_from_string(c_name.as_ptr()) };
+
+        if tmp.is_null() {
+            None
+        } else {
+            Some(FontDescription { pointer: tmp })
+        }
+    }
+
+    pub fn to_string(desc: &FontDescription) -> String {
+        unsafe {
+            let tmp = ffi::pango_font_description_to_string(desc.pointer);
+            debug_assert!(!tmp.is_null());
```

Comment:
The NULL check is handled by `from_glib_full` in case the return type is `String` and not `Option<String>`.